lib/libzfs_core/libzfs_core.c

```diff
@@ -686,8 +769,28 @@ lzc_send_resume_redacted(const char *snapname, const char *from, int fd,
 	if (redactbook != NULL)
 		fnvlist_add_string(args, "redactbook", redactbook);
 
+	sendargs.inputfd = pipefd[0];
+	sendargs.outputfd = fd;
+	sendargs.ioctlfd = pipefd[1];
+
+	pthread_create(&mythread, NULL, __do_send_output, (void *)&sendargs);
+
 	err = lzc_ioctl(ZFS_IOC_SEND_NEW, snapname, args, NULL);
+
+	close(pipefd[1]);
+
+	pthread_join(mythread, (void *)&dumbstatus);
+
 	nvlist_free(args);
+
+
+	if (dumbstatus != 0) {
+		err = dumbstatus;
+		// this seems really rude but what're ya gonna do
+		errno = dumbstatus;
```

Comment:
Why does errno need to be set here?

Reply:
I'll add a comment about this, but without it, you end up with strange outcomes when errno would be set in the writer thread where tests that pass with just this change error out with "internal error: [...]: Success" instead. Examples include zfs_receive_to_encrypted and send_encrypted_props (though not every time).

Reply:
Comment added; please let me know if you need anything further about this!